### ISSUE TITLE
~/aiidalab folder owned by wrong user

### ISSUE THREAD

**Issue by giovannipizzi** [aiidalab/aiidalab-launch#72]
On my computer my user has ID 1002, while the files are owned by user 1001. So I cannot access some of them, and my backup tools that backup the home complain and give errors. 
One more complex option if we really want to have these files in the home is to bind-mount /etc/passwd and /etc/group as read-only, and in the docker run specify the user. But it's not obvious to make it work correctly. I feel it's better to put these files not on a bind-mounted folder, but inside a docker volume?

**Comment by csadorf** [comment]
Thanks for reporting this issue and this is actually a problem that has been brought up before and there is actually no generally applicable solution to this problem within the docker ecosystem to my knowledge. 

Here are the two approaches to workaround the problem in this specific context:

1. For development: it is recommended to use an IDE like vscode, see the first tip [here](https://aiidalab.readthedocs.io/en/latest/app_development/create.html).
2. For general usage: I think that AiiDAlab Launch should support the use of a volume instead of a bind mount. This would alleviate the permissions issues that you observe, but obviously also makes the data harder to transfer.

W.r.t. point 2: As a first measure, we could potentially detect the issue by directly checking the ID of the current user, and in this case recommend the use of a volume. I would prefer to avoid messing with `/etc/passwd` since I suspect that the additional complexity would outweigh the potential benefits. A potential alternative approach might be to set the ID of the aiida user within the container to the same as the host user.

**Comment by giovannipizzi** [comment]
Hi, checking the ID and warning is better than nothing, still I think there should be an aiidalab-launch option like `--use-volume` or similar to launch.

I'm wondering how often one accesses the data. If it's just to check it, we could think to have an `export` function that launches a simple container mounting the same volume and allows to copy out the data. However I imagine that there are cases where you want to change the data easily?

In this case probably your last suggestion is the best, even if also there there are technicalities.

E.g. is the user with the correct UID created in the image, or every time the container starts? In the first case, instead of running directly the container from the image on DockerHub, one can build a new image just with the creation of the user with the default username, and the correct UID; and then run with that. Maybe this is a bit better even if more convoluted. When new versions of AiiDAlab are released, one just needs to recreate this final image.
In the second case, one can pass the UID as a env var and generate it at launch time in one of the CMD run scripts. However I'm not sure it's a good idea, as this means that you cannot have AiiDAlab already installed as that user, but delay its installation for the user until the user has been created. Also, the CMD has to be run by root (but maybe this is already the case if we are using passenger phusion as the base image.

So I would go with the first approach (and then no need to use volumes).
What do you think?

**Comment by csadorf** [comment]
> So I would go with the first approach (and then no need to use volumes). What do you think?

I already implemented the possibility to use volumes (https://github.com/aiidalab/aiidalab-launch/pull/74), however, I am still working out the details for a reasonable migration path. Unfortunately I made the mistake of not saving the config in case that the user did not make any changes. However, this also means that I can't know whether aiidalab-launch was already used unless there is a container. I think I am simply going to assume that if a user already has an `aiidalab` directory, that they want to mount it. Otherwise, I will default to the use of a volume in the future.
